src/main/java/fr/quatrevieux/araknemu/game/player/characteristic/PlayerLife.java

```diff
@@ -46,7 +49,45 @@ public int max() {
 
     @Override
     public int current() {
-        return entity.life();
+        return entity.life() + calculateLifeRegeneration();
+    }
+
+    private int calculateLifeRegeneration() {
+        if (lifeRegeneration == 0) return 0;
+
+        long currentTime = System.currentTimeMillis();
+        int lifeToAdd = (int) ( currentTime - lifeRegeneration ) / lifeRegenerationSpeed;
+        int currentLife = this.entity.life();
+
+        if (this.max <= (lifeToAdd + currentLife)) {
+            lifeToAdd = this.max - currentLife;
+        }
+
+        return lifeToAdd;
+    }
+
+    /**
+     * calculate the life regeneration and set the lifeRegeneration timestamp to zero
+     */
+    public void stopLifeRegeneration() {
+        setLifeWithCurrentRegeneration();
+        lifeRegeneration = 0;
+    }
+
+    /**
+     * set the life to: entity.life() + calculateLifeRegeneration()
+     */
+    public void setLifeWithCurrentRegeneration() {
+        entity.setLife(current());
+    }
+
+    /**
+     * Set the lifeRegeneration timestamps to System.currentTimeMillis()
+     * @param lifeRegenerationSpeed The speed of the life regeneration
```

Comment:
Change description with : The required delay in milliseconds to regenerate 1 life point